core/clustersmngr/clustersmngr.go

```diff
@@ -130,6 +132,24 @@ func (cp *clientsPool) Add(cfg ClusterClientConfig, cluster Cluster) error {
 	return nil
 }
 
+// Add adds a cluster client to the clients pool with the given user impersonation
+func (cp *clientsPool) AddSync(cfg ClusterClientConfig, cluster Cluster, mutex *sync.Mutex) error {
+	config := cfg(cluster)
+
+	leafClient, err := client.New(config, client.Options{
+		Scheme: cp.scheme,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create leaf client: %w", err)
+	}
+
+	mutex.Lock()
```

Comment:
is it possible that `mutex` happens to be `nil` ?
if so, it would be nice to check
`if mutex == nil` and return an error before this line.

Reply:
I'm not sure I'm a fan of this API - it _can_ be used securely, but now any time anybody uses `AddSync`, everybody always has to use `AddSync`. Could we instead just put the mutex in the pool itself, and always lock?